hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/api/Matcher.kt

```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api
+
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import org.opensearch.index.query.AbstractQueryBuilder
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.Attributes
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes(
+  JsonSubTypes.Type(value = BooleanMatcher::class, name = "Boolean"),
+  JsonSubTypes.Type(value = DateMatcher::class, name = "Date"),
+  JsonSubTypes.Type(value = DateTimeMatcher::class, name = "DateTime"),
+  JsonSubTypes.Type(value = IntMatcher::class, name = "Int"),
+  JsonSubTypes.Type(value = StringMatcher::class, name = "String"),
+  JsonSubTypes.Type(value = PncMatcher::class, name = "PNC"),
+)
+interface Matcher {
```

Comment:
We now have a matcherthat isn't a TypeMatcher, so extract out a Matcher interface